include/avif/internal.h

```diff
@@ -29,6 +29,14 @@ extern "C" {
             return ERR;       \
     } while (0)
 
+// Forward any error to the caller now or continue execution.
+#define AVIF_CHECKRES(A)                          \
+    do {                                          \
+        const avifResult avifCheckedResult = (A); \
```

Comment:
Nit: avifCheckedResult => checkedResult

Because of libavif's naming convention, `avifCheckedResult` unfortunately looks like a global function or a type.

We can also use a shorter name such as `result__`. The trailing underscores help avoid the -Wshadow warnings.